vm/src/compile.rs

```diff
@@ -115,7 +115,7 @@ impl Compiler {
         self.emit(Instruction::ReturnValue);
     }
 
-    fn compile_statements(&mut self, statements: &Vec<ast::LocatedStatement>) {
+    fn compile_statements(&mut self, statements: &[ast::LocatedStatement]) {
```

Comment:
I assume this variation is better rust code, but still I feel like the Vec<bladibla> syntax is clearer. At least to me.

Reply:
This allows non-Vec data structures which can produce slices for free to call the function with less overhead.